document/src/main/java/com/ritense/document/autoconfigure/DocumentAutoConfiguration.java

```diff
@@ -184,15 +192,22 @@ public JsonSchemaDocumentDefinitionResource documentDefinitionResource(
     @ConditionalOnMissingBean(DocumentResource.class)
     public JsonSchemaDocumentResource documentResource(
         DocumentService documentService,
-        DocumentDefinitionService documentDefinitionService
+        DocumentDefinitionService documentDefinitionService,
+        TenantResolver tenantResolver
     ) {
-        return new JsonSchemaDocumentResource(documentService, documentDefinitionService);
+        return new JsonSchemaDocumentResource(documentService, documentDefinitionService, tenantResolver);
     }
 
     @Bean
     @ConditionalOnMissingBean(DocumentSearchResource.class)
-    public JsonSchemaDocumentSearchResource documentSearchResource(DocumentSearchService documentSearchService) {
-        return new JsonSchemaDocumentSearchResource(documentSearchService);
+    public JsonSchemaDocumentSearchResource documentSearchResource(
+        DocumentSearchService documentSearchService,
```

Comment:
For other beans the constructor arguments are final so it seems. shouldn't these be final as well then to match chosen coding style